docs/user_guide/models/cellassign.rst

```diff
@@ -0,0 +1,169 @@
+draw, :math:`\bar{\delta}` is initialized to 0, :math:`\sigma^2` is initialized to 1, :math:`\pi_i` is
+initialized to :math:`1/C`, :math:`\beta_{gi}` is initialized with a :math:`\mathcal{N}(0, 1)` draw,
+and the inverse dispersion parameter :math:`\log a_i` is initialized to zero. Additionally, :math:`b` is fixed a priori to be
+
+.. math::
+    b = \frac{1}{2(v_2 - v_1)^2}, \tag{7}
+
+where :math:`v_1` and :math:`v_2` are the first and second center determined as stated previously.
+
+E step
+-------
+
+The E step consists of computing the expected joint log likelihood with respect to the conditional posterior,
+:math:`p(z_n \mid x_n, \delta, \beta, a, \pi)`, for each cell. This conditional posterior is
+computed in closed form as
+
+.. math::
+
+    \gamma_{nc} := p(z_n = c \mid x_n, \delta, \beta, a, \pi) \propto p(z_n = c \mid \pi)\prod_g p(x_{ng} \mid \mu_{ngc}, \tilde{\phi}_{ngc}), \tag{8}
+
+which is normalized over all $c$. The expected joint log likelihood over all :math:`N` cells is then computed as
+
+.. math::
+   :nowrap:
+
+    \begin{align}
+        \begin{split}
+        \mathbb{E}_{z \mid X,\pi,\delta, \beta, a}[\log p(X, \pi, \delta \mid \beta, a, \bar{\delta}, \sigma^2)]
+        % &=\log p(\theta) + \log p(\delta) +\sum_n\sum_{c}\gamma_{nc}\log p(y_{n}|c)\\
+        &= \sum_{n=1}^{N}\sum_{c=1}^{C}\gamma_{nc}\sum_{g=1}^{G}\log p(x_{ng}|z_n = c) \\
+        & \qquad + \log p(\pi) + \log p(\delta).
+        \end{split} \tag{9}
+    \end{align}
+
+Herein lies the major difference between the scvi-tools implementation and the original CellAssign implementation.
+Notably, in scvi-tools we compute this expected joint log likelihood using a mini-batch of 1,024 cells, using the fact that
+
+.. math::
+
+    \sum_{n=1}^{N}\sum_{c=1}^{C}\gamma_{nc}\sum_{g=1}^{G}\log p(x_{ng}|z_n = c) \approx \frac{N}{M}\sum_{m=1}^M\sum_{c=1}^{C}\gamma_{nc}\sum_{g=1}^{G}\log p(x_{\tau(m)g}|z_n = c) \tag{10}
+
+for a minibatch of $M<N$ cells, where :math:`\tau` is a function describing a permutation of the data indices $\{1, 2, ..., N\}$.
+
+M step
+-------
+
+The parameters of the expected joint log likelihood (:math:`\pi, \delta, \beta, a, \bar{\delta}, \sigma^2`) are optimized as
+in the original implementation [#ref2]_, using the Adam optimizer [#ref3]_, except that now an optimization step corresponds to data from one minibatch. Following the original implementation, we
+also clamped :math:`\delta > 2`. We also added early stopping with respect to the log likelihood of a held-out validation set.
+
+Tasks
+=====
+
+Cell type prediction
+---------------------
+
+The primary task of CellAssign is to predict cell types for each cell. This is accomplised by::
+
+    >>> model = CellAssign(adata, marker_gene_matrix, size_factor_key='size_factor')
+    >>> model.train()
+    >>> predictions = model.predict(adata)
+
+where `predictions` stores :math:`\gamma_{nc}` for each cell :math:`n` and cell type :math:`c`.
+
+
+Implementation details
```

Comment:
Given the number of variables here, I think a table like in the scVI guide would be helpful.